### PR TITLE
Exponential backoff for pod creation

### PR DESCRIPTION
**Pull request by jveski**
If a composition is impossible to synthesize, we should eventually slow down the synthesis loop to avoid the cost of starting/running the pods.

### CODE REVIEW

File: internal/controllers/synthesis/integration_test.go [hunk 1]
```diff
@@ -321,3 +321,89 @@ func TestControllerSwitchingSynthesizers(t *testing.T) {
 		assert.NotEqual(t, comp.Status.CurrentSynthesis.ResourceSlices, initialSlices)
 	})
 }
+
+func TestControllerBackoff(t *testing.T) {
```
Comment (AYM1607) [inline internal/controllers/synthesis/integration_test.go:325]: Should we split these off as well? Maybe pod_backoff_test?

File: internal/controllers/synthesis/lifecycle.go [hunk 2]
```diff
@@ -185,6 +185,17 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil
 	}
 
+	// Back off to avoid constantly re-synthesizing impossible compositions (unlikely but possible)
+	if current := comp.Status.CurrentSynthesis; current != nil && current.Attempts > 0 && current.PodCreation != nil {
+		const base = time.Millisecond * 250
```
Comment (AYM1607) [inline internal/controllers/synthesis/lifecycle.go:190]: Make configurable?

File: internal/controllers/synthesis/integration_test.go (hunk 1 above)
Comment (jveski) [reply internal/controllers/synthesis/integration_test.go:325]: I like it - done

**Review by AYM1607 — APPROVED** [review]
LGTM

**Pull request merged**